### DISCUSSION TITLE
A good guide for training a specific person in Dreambooth?

### DISCUSSION THREAD

**Discussion by Jonseed** [d8ahazard/sd_dreambooth_extension#440] [Q&A]
I have been testing TI and Dreambooth training for the last several weeks, and I have yet to produce a model or embedding that realistically and reliably reproduces a specific person, like me or my wife. I'm not sure what I'm doing wrong. Are there any good guides as to how to do this well?

**Comment by d8ahazard** [comment]
IDK about a "good" guide, but you can probably try something like so:

More images means more editability. Make sure they aren't blurry, try to avoid cropping parts of the face, stuff like that.

Go for a mix of face closeups, torso+face, and full body for best results. Appx 5-10 each should be enough, but again, more is better, IMHO.

Captions. I think they help a lot, as well as training with text encoder. You can use my "Smart Process" extension to do a rough job of captioning images, then go through and fix things if they're way off base. Ensure your class word is in all of the prompts. So, for your wife, "woman", and for you, "man". I've seen others say using "person" is sufficient, but why not give the robot more data?

Prior preservation. ~10 class images per instance image. No more than 1k, unless you're just bored and want to watch your CPU spin. Use [filewords] to generate your captions based off of instance image prompts.

Steps/epochs - 80-100 steps per instance image, or just 80-100 epochs. ;)

EMA...IDK. I haven't been using it, and don't really notice a difference.

Otherwise, learning rate, scheduler, gradient steps - I mostly just leave that alone. 

Happy training. ;)

**Reply by Jonseed** [reply]
Thanks for these tips. I think I've been doing most of these things, and still not getting great results. I'll keep testing.

**Reply by Kahldris** [reply]
> Captions. I think they help a lot, as well as training with text encoder.

After captioning the dataset, may I ask what you've found best to the enter into the _Filewords_ and _Prompts_ fields under _Concept 1_? I struggle a little to understand what should go where. Say you're you're using `[filewords]` and training a `dog` with the unique token `sks`:

**Filewords fields**
- _Instance Token_: Should I only put `sks` here? Does it add the token to each images' caption, or is the token already meant to be in the image captions as well? If you already have the token at the start of your captions already, is adding it to this field even necessary?
- _Class Token_: Kind of same question as above - does entering `dog` here add it to the start of the captions? Should I enter it if I already have `dog` somewhere in every caption? I've seen some people not using a class token and thought maybe that's why.

---

**Prompts fields**
- _Instance Prompt_: Just put `[filewords]` here and nothing else? What if I put nothing in _Instance Token_ above, if I enter `sks [filewords]` in this field is it basically the same?
- _Class Prompt_: This is only for generating the class images before training correct? Kind of same question as above, should I only enter `[filewords]` here and leave the _Class Token_ field blank because `dog` is already in every caption?

---

- If I'm not using captions or `[filewords]` at all, I should leave the _Instance Token_ and _Class Token_ fields blank and only put `sks` under _Instance Prompt_ and `dog` under _Class Prompt_? Is that correct? I feel _Instance Token_ still sounds like the best field to put `sks`, but because it's under the _Filewords_ section and I'm not using `[filewords]`, that's not the case?

Hope I made those questions clear enough, I know there's no perfect answer. I've done lots of experimenting myself, but still not sure because I think my old training without captioning or using `[filewords]` is still better... feel like I'm missing something.

**Reply by alkah3st** [reply]
I have the same questions as above.

**Reply by a-l-e-x-d-s-9** [reply]
@Kahldris 
Have you figured out answers to your questions?

**Reply by alkah3st** [reply]
@Kahldris @a-l-e-x-d-s-9 

as an fyi, I did some experimenting here:

https://www.reddit.com/r/StableDiffusion/comments/zl6wig/is_there_a_good_tutorial_for_dreambooth_on/

I didn't test captions, but the only successful result is labeled Test #1. I tried many combinations of filling out these fields. It doesn't seem to matter how the prompt token is implemented, as far as I can tell.

**Reply by terrificdm** [reply]
> > Captions. I think they help a lot, as well as training with text encoder.
> 
> After captioning the dataset, may I ask what you've found best to the enter into the _Filewords_ and _Prompts_ fields under _Concept 1_? I struggle a little to understand what should go where. Say you're you're using `[filewords]` and training a `dog` with the unique token `sks`:
> 
> **Filewords fields**
> 
> * _Instance Token_: Should I only put `sks` here? Does it add the token to each images' caption, or is the token already meant to be in the image captions as well? If you already have the token at the start of your captions already, is adding it to this field even necessary?

_**This token will be added to each caption of the image**_

> * _Class Token_: Kind of same question as above - does entering `dog` here add it to the start of the captions? Should I enter it if I already have `dog` somewhere in every caption? I've seen some people not using a class token and thought maybe that's why.
> 
**_If there was a class word(e.g. 'dog' in your case) in each caption of the image, I don't think you need to put an extra class token 'dog' here_**

> **Prompts fields**
> 
> * _Instance Prompt_: Just put `[filewords]` here and nothing else? What if I put nothing in _Instance Token_ above, if I enter `sks [filewords]` in this field is it basically the same?

**_Just put `[filewords]` here, that's enough. And if you leave the Instance Token empty and put the unique word here like `sks [filewords]` you described, the result is the same_**

> * _Class Prompt_: This is only for generating the class images before training correct? Kind of same question as above, should I only enter `[filewords]` here and leave the _Class Token_ field blank because `dog` is already in every caption?
> 
**_Just put `[filewords]`, nothing else_** 

> * If I'm not using captions or `[filewords]` at all, I should leave the _Instance Token_ and _Class Token_ fields blank and only put `sks` under _Instance Prompt_ and `dog` under _Class Prompt_? Is that correct? I feel _Instance Token_ still sounds like the best field to put `sks`, but because it's under the _Filewords_ section and I'm not using `[filewords]`, that's not the case?
> 

**_Yes, If you don't plan to use [filewords], I believe that those two tokens won't be workable._** 

> Hope I made those questions clear enough, I know there's no perfect answer. I've done lots of experimenting myself, but still not sure because I think my old training without captioning or using `[filewords]` is still better... feel like I'm missing something.

**_Basically, your final prompt for your training stage of Dreambooth is the composition of tokens and prompt, 'instance token'+'class token'+'[filewords]'_**
